pyquil/api/_quantum_computer.py

```diff
@@ -144,6 +144,13 @@ def run(self, executable: Executable,
             are a list of floats or integers.
         :return: A numpy array of shape (trials, len(ro-register)) that contains 0s and 1s.
         """
+        # This prevents a common error where users expect QVM.run()
+        # and QPU.run() to be interchangeable. QPU.run() needs the
+        # supplied executable to have been compiled, QVM.run() does not.
```

Comment:
some more color here: if you construct a qvm *with a topology*, the code will assume you care about realism and *will* error if you pass a program instead of an executable. However, we thought that would be annoying if you didn't really care about compiling/realism so if you construct a {n}q-qvm without a topology it will accept `Program`s

```python
>> get_qc('2q-qvm').run(program) # fine
>> get_qc('Aspen-1-2Q-A-qvm').run(program) # error! you need to compile
```

this might be more confusing than it's worth